osiris// Osiris
// Copyright(C) 2019 Christoph GÃ¶rn
//
// This program is free software: you can redistribute it and / or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

// Protocol Buffers Specification of the Osiris API

syntax = "proto3";

package osiris;

// Interface exported by the server.
service Osiris {
  // Obtains API Info.
  rpc Info(Empty) returns (InfoResponse);
}

// The Info
message InfoResponse {
  // Version of the Osiris API
  string version = 1;

  // Version of connexion module used by Osiris API service
  string connexionVersion = 2;

  // Version of Jaeger Client used by Osiris API service
  string jaegerClientVersion = 3;
}

message Empty {};
